Unix/http/httpclientauth.c

```diff
@@ -728,6 +729,28 @@ static _Success_(return == 0) int _GssClientInitLibrary( _In_ void* data, _Outpt
         _g_gssClientState.Gss_Wrap   = (Gss_Wrap_Func) fn_handle;
 
 #if GSS_USE_IOV
+#if defined(macos)
+        fn_handle = dlsym(libhandle, "__ApplePrivate_gss_unwrap_iov");
```

Comment:
Mac is an interesting case in that it's a single platform with far less variability than other platforms. For example, you don't have a choice as to what libraries are installed; these tend to just come with the system.

Given this, would it make sense to statically link against MacOS libraries, or dynamically link against them. I'm not saying what you're doing is wrong. I just want you to consider if this is the right approach or not. To be honest: I'm not sure. But I did want you to consider it.